Assignment1/assignment2/lib/utilities.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:assignment2/users.dart';
+import 'dart:collection';
+
+Map<int, User> currentSession = Map();
+
+void nameHandler(String name) {
+  if (name.length == 0 || name.contains('[0-9]')) {
+    throw Exception(
+        "Name entered isn't valid, please enter appropriate details");
```

Comment:
Instead of creating a generic Exception with a custom message, a custom exception class can be created by extending the Exception class and used. This will ensure more robust exception handling.